src/pytest/__init__.py

```diff
@@ -95,5 +94,9 @@
 ]
 
 
-_setup_collect_fakemodule()
-del _setup_collect_fakemodule
+def __getattr__(name):
+    if name == "collect":
+        from _pytest.compat import _setup_collect_fakemodule
```

Comment:
Perhaps follow the previous approach of using a real module (say `_pytest._collect`) and use it here?

```python
def __getattr__(name):
    if name == "collect":
        from _pytest import _collect as collect
        return collect
    raise `AttributeError(name)` 
```

Hmmm unfortunately `__getattr__` is [Python 3.7 only](https://www.python.org/dev/peps/pep-0562/)?

Reply:
I'd rather keep the original code as-is, given that it is supposed to not stay anyway.

Reply:
As a follow-up or inline, adding to sys. Modules and adding a deprecation warning subclass of the module type should improve the general situation

Reply:
Adding to sys.modules would likely break that test: https://github.com/pytest-dev/pytest/pull/6803/files#diff-dc953e3e35889673e6c7ba78247b25d9R49-R50

Reply:
The deprecation warning should be done in a follow-up.